src/components/commons/Buttons.tsx

```diff
@@ -1,24 +1,147 @@
 import { PropsWithChildren } from "react";
 import classNames from "classnames";
 
-/**
- * Container implementation that is part of the design language.
- */
-export function Button(
-  props: PropsWithChildren<{ className?: string; text: string }>
-): JSX.Element {
+export function Button({
+  buttonText,
+  className,
+  disabled,
+  href,
+  onClick,
+  isCTAButton = true,
+}: {
+  buttonText: string;
+  className?: string;
+  disabled?: boolean;
+  href?: string;
+  onClick?: () => void;
+  isCTAButton?: boolean;
+}) {
+  if (href) {
+    return (
+      <a
+        href={href}
+        className={classNames({ "pointer-events-none": disabled })}
+      >
+        <ButtonElement
+          buttonText={buttonText}
+          className={className}
+          disabled={disabled}
+          onClick={onClick}
+          isCTAButton={isCTAButton}
+        />
+      </a>
+    );
+  }
+  return (
+    <ButtonElement
+      buttonText={buttonText}
+      className={className}
+      disabled={disabled}
+      onClick={onClick}
+      isCTAButton={isCTAButton}
+    />
+  );
+}
+
+function ButtonElement({
+  buttonText,
+  className,
+  disabled,
+  onClick,
+  isCTAButton = true,
+}: {
+  buttonText: string;
+  className?: string;
+  disabled?: boolean;
+  onClick?: () => void;
+  isCTAButton?: boolean;
+}) {
   return (
     <button
+      type="button"
+      onClick={onClick}
+      disabled={disabled}
       className={classNames(
-        "button rounded border border-primary-500 py-2 px-6 font-medium text-white bg-primary-500 hover:bg-primary-600",
-        props.className
+        className ?? "py-4 px-28",
```

Comment:
dont think we should use horizontal padding, should set min width

Reply:
for height would it be fine to leave as py-4 if custom className is provided then it would be using custom classname's vertical padding or should we set min-height too